File: domain/src/main/java/com/github/dkurata38/liblio/domain/user/UserId.java
package com.github.dkurata38.liblio.domain.user;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NonNull;
import lombok.ToString;

import java.util.UUID;

@EqualsAndHashCode
@ToString
public class UserId {
    @Getter
    private final String value;

    public UserId(@NonNull String value) {
        this.value = value;
    }

    public static UserId generate() {
        return new UserId(UUID.randomUUID().toString());
    }
}

File: domain/src/main/kotlin/com/github/dkurata38/liblio/domain/user/UserId.kt
package com.github.dkurata38.liblio.domain.user

import java.util.*

data class UserId(val value: String) {

    init {
        require(value.isNotEmpty())
    }

    companion object {
        @JvmStatic
        fun generate(): UserId {
            return UserId(UUID.randomUUID().toString())
        }
    }
}

